src/app/components/shared/datetime/datetime/datetime.component.ts

```diff
@@ -0,0 +1,132 @@
+    // eg. without this condition, setting both dateOnly and timeOnly to true would result in the dateOnly format being used
+    // which is not obvious to the programmer without looking at the code
+    if (
+      isInstantiated(this.dateOnly) &&
+      this.dateOnly !== false &&
+      isInstantiated(this.timeOnly) &&
+      this.timeOnly !== false
+    ) {
+      return "yyyy-MM-dd HH:mm:ss";
+    }
+
+    // we use isInstantiated() to check if the humanized attribute is set
+    // and we also check that it is not set to false
+    // by allowing boolean values, we support reactive formats (e.g. [dateOnly]="isDateOnly")
+    if (isInstantiated(this.dateOnly) && this.dateOnly !== false) {
+      return "yyyy-MM-dd";
+    } else if (isInstantiated(this.timeOnly) && this.timeOnly !== false) {
+      return "HH:mm:ss";
+    }
+
+    // include both date and time if neither dateOnly or timeOnly are specified
+    return "yyyy-MM-dd HH:mm:ss";
+  }
+
+  // the longTimezone() method returns the timezone name and offset
+  // eg. "Australia/Perth +08:00"
+  // we use this method to display the full timezone name and offset in the tooltip
+  private longTimezone(): string {
+    let zoneObject: Zone;
+
+    // if the timezone is not set (either implicitly or explicitly)
+    // the datetime will be localized to the users local timezone
+    if (typeof this.timezone === "undefined") {
+      zoneObject = DateTime.local().zone;
+    } else {
+      zoneObject =
+        this.timezone instanceof Zone
+          ? this.timezone
+          : IANAZone.create(this.timezone["identifier"] ?? this.timezone);
+    }
+
+    const timezoneName = zoneObject.name;
+    const offsetValue = zoneObject.formatOffset(
+      this.luxonDateTime.offset,
+      "short"
+    );
+
+    // if we don't know the timezone name, timezoneName will be undefined, and we should just return the offset
+    // this can happen because valid IANAZone objects can be created from offsets
+    // eg. [timezone]="utc+08:00" will create a valid IANAZone object with the timezone.name property name "utc+08:00"
+    // as this is not an actual timezone name (and is actually an offset), we should not emit it in the
+    // in place of the timezone name
+    if (!timezoneName || !this.isIanaTimezone(timezoneName)) {
+      return offsetValue;
+    }
+
+    return `${timezoneName} ${offsetValue}`;
+  }
+
+
+  // it is possible to provide a offset as Luxon Zone (not an IANA timezone)
+  // see: https://github.com/moment/luxon/blob/3125686af82d9a25c7267a1cf1eb838a3d41144f/docs/zones.md?plain=1#L55-L58
+  // this is an incorrect timezoneName, as it is an offset, not timezone name. We should therefore omit the
+  // timezoneName from the tooltip in this case
+  private isIanaTimezone(timezoneName: string): boolean {
+    return !(
+      timezoneName.startsWith("UTC") ||
+      timezoneName === "utc" ||
+      timezoneName === "system" ||
+      timezoneName === "local"
+    );
+  }
+}
```

Comment:
this logic seems overly complex and also not related to this class?

Reply:
I've moved most of the timezone logic to the `zoned-datetime.component.ts`. However, since the `datetime.component.ts` and `duration.component.ts` both require the users local timezone, I'm moving methods to get the users local timezone to the abstract template coponent

Reply:
ditto above